operator/pkg/lbipam/lbipam_fixture_test.go

```diff
@@ -0,0 +1,353 @@
+	}
+	fixture.ciliumCS.ReactionChain = append([]k8s_testing.Reactor{poolReactor}, fixture.ciliumCS.ReactionChain...)
+
+	// Create a new mocked core client set
+	fixture.coreCS = fake.NewSimpleClientset()
+	// / Create a new reactor, any API call related to services made to the mocked client will trigger
+	// our `fixture.svcReactor` callback if set.
+	svcReactor := &k8s_testing.SimpleReactor{
+		Verb:     "*",
+		Resource: "services",
+		Reaction: func(action k8s_testing.Action) (handled bool, ret runtime.Object, err error) {
+			if fixture.svcReactor != nil {
+				fixture.svcReactor(action)
+			}
+			return false, nil, nil
+		},
+	}
+	fixture.coreCS.ReactionChain = append([]k8s_testing.Reactor{svcReactor}, fixture.coreCS.ReactionChain...)
+
+	// Get a pool and service client from the sets.
+	fixture.poolClient = fixture.ciliumCS.CiliumV2alpha1().CiliumLoadBalancerIPPools()
+	fixture.svcClient = fixture.coreCS.CoreV1()
+
+	// This type is used to wire the a init callback to the LB-IPAM cell, which we only need during testing.
+	type InitDoneCallback struct {
+		cell.Out
+
+		Callback LBIPAMInitDone `group:"lb-ipam-init-ready"`
+	}
+
+	// Construct a new Hive with mocked out dependency cells.
+	fixture.hive = hive.New(
+		// Create a resource from the mocked clients
+		cell.Provide(func(lc hive.Lifecycle, c k8sClient.Clientset) resource.Resource[*cilium_api_v2alpha1.CiliumLoadBalancerIPPool] {
+			return resource.New[*cilium_api_v2alpha1.CiliumLoadBalancerIPPool](
+				lc, utils.ListerWatcherFromTyped[*cilium_api_v2alpha1.CiliumLoadBalancerIPPoolList](
+					c.CiliumV2alpha1().CiliumLoadBalancerIPPools(),
+				),
+			)
+		}),
+		cell.Provide(func(lc hive.Lifecycle, c k8sClient.Clientset) resource.Resource[*core_v1.Service] {
+			return resource.New[*core_v1.Service](
+				lc, utils.ListerWatcherFromTyped[*core_v1.ServiceList](
+					c.CoreV1().Services(""),
+				),
+			)
+		}),
+
+		// Provide the mocked client cells directly
+		cell.Provide(func() k8sClient.Clientset {
+			return &k8sClient.FakeClientset{
+				KubernetesFakeClientset: fixture.coreCS,
+				CiliumFakeClientset:     fixture.ciliumCS,
+			}
+		}),
+
+		// Provide a init done callback, to be triggered when the LB-IPAM is done initializing.
+		cell.Provide(func() InitDoneCallback {
+			return InitDoneCallback{
+				Callback: initDone,
+			}
+		}),
+
+		// Provide an operator config, with the BGP LB announcement flag set, which causes the BGP Control plane
+		// LB Class to be added.
+		cell.Provide(func() *operatorOption.OperatorConfig {
+			return &operatorOption.OperatorConfig{
+				BGPAnnounceLBIP: true,
+			}
+		}),
+
+		// This callback will write the LB-IPAM instance to the fixture so internals can be checked in the tests.
+		cell.Invoke(func(lbIPAM *LBIPAM) {
+			fixture.lbIPAM = lbIPAM
+		}),
+
+		// Add the actual LB-IPAM cell under test.
+		Cell,
+	)
+
+	// Set the LB-IPAM enable flag.
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	fixture.hive.RegisterFlags(flags)
```

Comment:
You can also do `hive.Viper().Set("lb-ipam", "true")`

Reply:
Flag has been removed in favor of starting and stopping the controller dynamically